src/Service/Telemetry/MyTelemetryInitializer.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using Azure.DataApiBuilder.Product;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+
+public class MyTelemetryInitializer : ITelemetryInitializer
+{
+    public static readonly IReadOnlyDictionary<string, string> GlobalProperties = new Dictionary<string, string>
+    {
+        { "UserAgent", $"{ProductInfo.GetDataApiBuilderUserAgent()}" },
+        // Add more custom properties here
+    };
+
+    public void Initialize(ITelemetry telemetry)
+    {
+        telemetry.Context.Cloud.RoleName = "DataApiBuilder";
```

Comment:
Since we hardcode the role name, do we need to recommend that customers have one appinsights instance per Data API builder deployment?

Reply:
updated it to not use hardcodded value, rather use dab_oss_version/dab_hosted_version accordingly based on oss/hosted scenario resp.